Mathlib/RingTheory/GradedAlgebra/Basic.lean

```diff
@@ -307,6 +307,31 @@ def GradedRing.projZeroRingHom : A →+* A where
       simp only [add_mul, decompose_add, add_apply, AddMemClass.coe_add, ha, hb]
 #align graded_ring.proj_zero_ring_hom GradedRing.projZeroRingHom
 
+section GradeZero
+
+/--
+The semiring homomorphism from `A` to `𝒜 0` sending every `a : A` to `a₀`.
+-/
+def GradedRing.projZeroRingHom' : A →+* 𝒜 0 :=
+  ((GradedRing.projZeroRingHom 𝒜).codRestrict _ fun _x => SetLike.coe_mem _ :
+  A →+* GradeZero.subsemiring 𝒜)
+
+@[simp] lemma GradedRing.coe_projZeroRingHom'_apply (a : A) :
+    (GradedRing.projZeroRingHom' 𝒜 a : A) = GradedRing.projZeroRingHom 𝒜 a := rfl
+
+lemma GradedRing.projZeroRingHom'_apply_coe (a : 𝒜 0) :
```

Comment:
Does the linter complain if you mark this `simp`?

Reply:
Hi Eric! I see some of the error messages look like this:
```
error: SetLike.GradeZero.coe_natCast.{u_4, u_2, u_1} Left-hand side does not simplify, when using the simp lemma on itself.
```
Does it mean that I need to delete the `simp` tag of the theorems?